### PR TITLE
macOS Core Video fix

### PR DESCRIPTION
**Pull request by steventang-gpsw**
This is the replacement of PR#508, took review comments of PR # 508 into consideration.

### CODE REVIEW

File: libnodegl/backends/gl/gpu_ctx_gl.h [hunk 1]
```diff
@@ -63,6 +65,9 @@ struct gpu_ctx_gl {
 #if defined(TARGET_IPHONE)
     CVPixelBufferRef capture_cvbuffer;
     CVOpenGLESTextureRef capture_cvtexture;
+#elif defined(TARGET_DARWIN)
+	CVPixelBufferRef capture_cvbuffer;
```
Comment (kgrant-gpsw) [inline libnodegl/backends/gl/gpu_ctx_gl.h:69]:
```suggestion
    CVPixelBufferRef capture_cvbuffer;
```

File: libnodegl/backends/gl/gpu_ctx_gl.c [hunk 2]
```diff
@@ -131,19 +132,74 @@ static int wrap_capture_cvpixelbuffer(struct gpu_ctx *s,
...
+                                                                 &cv_texture);
+    if (cv_ret != kCVReturnSuccess) {
+        LOG(ERROR, "could not create CoreVideo texture from CVPixelBuffer: %d", cv_ret);
+        return NGL_ERROR_EXTERNAL;
+    }
+    
+    GLuint fbName = CVOpenGLTextureGetName(cv_texture);
+    ngli_glBindTexture(gl, GL_TEXTURE_RECTANGLE, fbName);
+    ngli_glTexParameteri(gl, GL_TEXTURE_RECTANGLE, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+    ngli_glTexParameteri(gl, GL_TEXTURE_RECTANGLE, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
```
Comment (kgrant-gpsw) [inline libnodegl/backends/gl/gpu_ctx_gl.c:162]: Do we need `ngli_glBindTexture(gl, GL_TEXTURE_RECTANGLE, 0)` here?

File: libnodegl/backends/gl/gpu_ctx_gl.h [hunk 3]
```diff
@@ -41,6 +41,8 @@ struct rendertarget;
 
 typedef void (*capture_func_type)(struct gpu_ctx *s);
 
+#define SAFE_CFRELEASE(p) { if ( (p) ) { CFRelease(p); (p) = NULL; } }
```
Comment (kgrant-gpsw) [inline libnodegl/backends/gl/gpu_ctx_gl.h:44]: A `do { ... } while (0)` loop works in even more places than a `{ ... }` block in a macro.

**Review by kgrant-gpsw — COMMENTED** [review]
Looks OK but I suggested a few minor changes, please see comments.

File: libnodegl/backends/gl/gpu_ctx_gl.h (hunk 3 above)
Comment (steventang-gpsw) [reply libnodegl/backends/gl/gpu_ctx_gl.h:44]: This is for dealing with those APIs like CVOpenGLTextureRelease, CVOpenGLESTextureRelease, those APIs are equivalent to CFRelease but with NULL check, anyway those APIs have cross-platform 
 issue. do {...} while (0) might need more updates, I'll look into it to check if it is safe to update.

**Comment by steventang-gpsw** [comment]
@cbadic-gpsw could you give me hint on how to fix the linter black checking failure? my understanding is that black is for formatting Python code, we don't have any Python code updated in this PR.

**Comment by cboesch-gpsw** [comment]
Please squash your changes and rebase your branch on `main`

**Comment by wmacdonald-gpsw** [comment]
Kevin has already covered the spots that I wanted to comment on. I think we just need to figure out the linter issue and squash/rebase.

**Comment by jbnoel-gpsw** [comment]
@kgrant-gpsw @steventang-gpsw This PR is not ready. I'm moving it to draft.
Please make sure your git history is clean before resubmitting the PR.
Check with @cboesch-gpsw if you need more details around the PR process.

**Comment by steventang-gpsw** [comment]
Sorry, since the directory structure changed, for easier to review, I created a new PR
https://github.com/gopro/gopro-lib-node.gl/pull/520
for replacing this one.

**Pull request closed without merging**